ppr-ui/src/components/collateral/Collateral.vue

```diff
@@ -192,6 +193,18 @@ export default defineComponent({
       }
     })
 
+    onUnmounted(() => {
+      // clear general collateral description if there is no valid text left in editor (html tags not included)
+      if (getAddCollateral.value.generalCollateral[0]?.description?.replace(/(<([^>]+)>)/ig, '').length === 0) {
+        setGeneralCollateral([])
+        // clear collateral step check mark if there are no vehicles
+        // (this resets check mark that was set by general collateral description)
+        if (getAddCollateral.value.vehicleCollateral.length === 0) {
+          getAddCollateral.value.valid = false
+        }
+      }
+    })
+
```

Comment:
Very cool, just curious, perhaps you already tried this as well but what if we made this conditional a computed property(so a reactive boolean in this case) and we set up a watcher on that property to clear the store?

Also, with regards to the `getAddCollateral.value.valid` is this computed currently based on have some char length or set manually by ui?

Reply:
I'm just a little wary of using unmounted lifecycle hook in a SPA, although i'm sure it's probably fine here..